velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -385,7 +382,15 @@ HiveDataSink::HiveDataSink(
                     inputType)
               : nullptr,
           getPartitionChannels(insertTableHandle),
-          nullptr) {}
+          getNonPartitionChannels(insertTableHandle),
+          !getPartitionChannels(insertTableHandle).empty()
```

Comment:
It seems sub-optimal to call getPartitionChannels(insertTableHandle) twice and create a vector just to check if it is empty